spec/fortnox/api/repositories/examples/only.rb

```diff
@@ -0,0 +1,42 @@
+
+shared_examples_for '.only' do |matching_filter, missing_filter|
+  describe '.only' do
+
+    let( :repository ){ described_class.new }
+    let( :vcr_dir ){ repository.options.json_collection_wrapper.downcase }
+
+    context "with no matches" do
+      subject do
+        VCR.use_cassette( "#{vcr_dir}/filter_miss" ) do
+          repository.only( missing_filter )
+        end
+      end
+
+      it{ is_expected.to be_instance_of( Array ) }
+      it{ is_expected.to have(0).entries }
+    end
+
+    context "with one match" do
+      subject do
+        VCR.use_cassette( "#{vcr_dir}/filter_hit" ) do
+          repository.only( matching_filter )
+        end
+      end
+
+      it{ is_expected.to be_instance_of( Array ) }
+      it{ is_expected.to have(1).entries }
+    end
+
+    context "with invalid filter" do
+      subject do
+        ->{
```

Comment:
That is not possible since the `raise` I'm trying to test actually occurs when the subject is created. What you could do is stick what is in the proc into the `expect` block instead, but that kind of ruins the symmetry of the tests and makes it more difficult to read since this one will be formatted differently for the others ...